### PR TITLE
creation of 'dnssec' module for configuring/enabling DNSSEC signing for a domain/hosted zone

### PR DESCRIPTION
**Pull request by bleachbyte**
This module has been moved into this repo as part of Login.gov's continuing effort to provide Terraform modules that we use to the open-source community.

Detailed information, including available variables, implementation examples, and recommended steps for effective key-signing key (KSK) rotation, is available in the `README` file within the module directory.

### CODE REVIEW

File: dnssec/main.tf [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        "kms:ScheduleKeyDeletion",
+      ]
+      resources = [
+        aws_kms_alias.dnssec[statement.key].target_key_arn
+      ]
+    }
+  }
+}
+
+resource "aws_iam_policy" "dnssec_disable_prevent" {
```
Comment (pauldoomgov) [inline dnssec/main.tf:206]: How do you have a special "I can smack around DNSSEC" role if this Deny is set inside a module?   Can we move the guardrail out to `all` like our RDS delete prevention guardrail?  If we need to ref resources we can do it with `data`, and only the handful of things we care about.  (login.gov, identitysandbox.gov)

**Review by pauldoomgov — CHANGES_REQUESTED** [review]
Comment inline.  Main issue - This hides the guardrail in a place that is hard to find, and makes excluding a special "break glass" role hard.

File: dnssec/main.tf (hunk 1 above)
Comment (bleachbyte) [reply dnssec/main.tf:206]: The IAM policy itself can be created on an optional basis (using the `protect_resources` flag), so its creation can be skipped entirely.

Additionally, this is a policy that gets attached to one or more roles -- meaning that we can select which roles do use it vs. which ones do not. That's how a "break glass" role would work in this case -- you simply create a role which doesn't have this policy attached.

**Review by pauldoomgov — APPROVED** [review]
LGTM!

**Pull request merged**